website/docs: stages cleanup (#21558)
goauthentik/authentik: website/docs/add-secure-apps/flows-stages/stages/authenticator_sms/index.md

---
title: SMS authenticator setup stage
---

The SMS Authenticator Setup stage enrolls an SMS-based authenticator for the current user by using either Twilio or a generic HTTP endpoint.

## Overview

> This stage stores a phone-based authenticator device for the user and sends one-time codes over SMS.

Reviewer: ```suggestion
This stage stores a phone number, enabling one-time codes to be sent via SMS.
```

> In normal mode, the enrolled device can later be used with the [Authenticator Validation stage](../authenticator_validate/index.md). In **verify only** mode, the stage only verifies ownership of a phone number during enrollment and stores a hash instead of the number itself.

Reviewer: ```suggestion
In normal mode, the enrolled phone number can later be used with the [Authenticator Validation stage](../authenticator_validate/index.md). In **verify only** mode, the stage only verifies ownership of a phone number during enrollment and stores a hash instead of the number itself.
```

- **Account SID / External API URL**: for Twilio this is the account SID; for the generic provider this is the target API URL.
- **Auth / token**: for Twilio this is the auth token; for the generic provider this is the bearer token or basic-auth username.
- **Auth password**: optional password for generic basic authentication.
- **Auth type**: choose **Basic** or **Bearer** authentication for the generic provider.
- **Verify only**: verify phone ownership during enrollment without storing the plain phone number for later MFA use.
- **Mapping**: optional webhook mapping used to customize the payload sent to custom providers.
- **Authenticator type name**: optional friendly name shown to the user in self-service settings.
- **Configuration flow**: optional authenticated flow that lets users enroll this authenticator from user settings.

## Flow integration

Use this stage in an enrollment or user-settings flow where the user should add an SMS authenticator.

To require SMS during login, add an [Authenticator Validation stage](../authenticator_validate/index.md) to the authentication flow and allow the **SMS** device class.

> If you enable **Verify only**, devices enrolled through this stage cannot be used by the Authenticator Validation stage.

Reviewer: ```suggestion
If you enable **Verify only**, phone numbers enrolled through this stage cannot be used by the Authenticator Validation stage.
```

Or SMS authenticators

Author: 👍
